quick-start/src/test/java/com/marklogic/quickstart/integrationtests/IntegrationUtils.java

```diff
@@ -0,0 +1,292 @@
+package com.marklogic.quickstart.integrationtests;
+
+import static io.restassured.RestAssured.given;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.marklogic.hub.HubTestBase;
+import com.marklogic.hub.flow.CodeFormat;
+import com.marklogic.hub.flow.DataFormat;
+import com.marklogic.hub.flow.FlowType;
+import com.marklogic.quickstart.auth.LoginInfo;
+import com.marklogic.quickstart.model.FlowModel;
+import com.marklogic.quickstart.model.JobQuery;
+
+import io.restassured.http.ContentType;
+import io.restassured.http.Cookie;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+
+public class IntegrationUtils {
+	
+	private String sessionID;
+	private String projectPath;
+	
+	// created this field to enable fetch the flowName that was created during the createFlow process
+	// and verify it.
+	private String flowName;
+	private int hmFlowNameCount = 1;
+	
+	private Cookie requestCookie;
+	private LoginInfo loginInfo = new LoginInfo();
+	private FlowModel flowModel;
+	private JobQuery jobQuery;
+	
+	// Tests api's to initialize the project folder and set the environment configuration
+	// Verifies if this the only project and status code of request is 200 OK
+	public Response initilizeProjectConfiguration() {
+		projectPath = new File(HubTestBase.PROJECT_PATH).getAbsolutePath();
+		Response projectInitResponse = 
+				given()
+					.contentType("application/x-www-form-urlencoded")
+					.queryParam("path", projectPath)
+				.when()
+					.post("/api/projects/");
+		
+		createLoginCredentials(projectInitResponse);
+		
+		return projectInitResponse;
+	}
+	// Tests api's to login into the quick-start application. These tests are using the 
+	// admin/admin credentials. A session cookie is received in header which has to be
+	// attached to the subsequent requests for security.
+	public String doLogin() {
+		Response loginResponse = 
+				given()
+					.body(loginInfo)
+				.when()
+					.post("/api/login");
+		if(loginResponse.statusCode() == 200) {
+			sessionID = StringUtils.substringBetween(loginResponse.getHeader("Set-Cookie"), "JSESSIONID=", ";");
+		}
+		
+		// building cookie to use for subsequent endpoint requests
+		buildCookie();
```

Comment:
so this relies on object state here -- you need that cookie to be around for subsequent things.  Makes me think this is not a utility class but something that holds state for a test session.

Reply:
I wanted to maintain the cookie as a private instance variable instead of passing around every time I invoke a method. Same with other instance variables as well. 

If this is not a good coding practice, I will pass the parameters and make the utils class static

Reply:
I have renamed the class to E2ETestsRequestHelper.Java